### PR TITLE
Adjustments to retry when setting passive GW on leadership lost

### PR DESCRIPTION
**Pull request by tpantelis**
Follow-up to https://github.com/submariner-io/submariner/pull/3016.

Refactor the `SetHAStatus` retry code in `onStartedLeading` to a function for reuse in `onStoppedLeading`. Modify it to retry on transient errors.

Depends on https://github.com/submariner-io/admiral/pull/919

### CODE REVIEW

**Review by aswinsuryan — APPROVED** [review]

File: pkg/gateway/gateway.go [hunk 1]
```diff
@@ -340,22 +329,44 @@ func (g *gatewayType) onStoppedLeading() {
...
-		}
 
-		return true, nil
-	}); err != nil {
-		logger.Errorf(err, "Failed updating pod label to passive: %s", err)
-	}
+	g.updateGatewayHAStatus(ctx, subv1.HAStatusPassive)
 
-	err := g.startLeaderElection(context.Background())
+	err := g.startLeaderElection(ctx)
```
Comment (vthapar) [inline pkg/gateway/gateway.go:335]: Why this change?

**Review by vthapar — APPROVED** [review]

**Review by yboaron — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]
